FlaskServer.py

```diff
@@ -41,9 +41,54 @@ def get_main_id():
     return jsonify(jsonResponse)
 
 @app.route('/api/all_objects', methods=['GET'])
-def get_all_objects():
-    jsonResponse = []
-    time = request.args.get('time')
+def handle_get_objects_request():
+    objects = get_all_objects(request.args.get("time"))
+    return jsonify(objects)
+
+@app.route('/<path:filename>', methods=['GET'])
+def get_file(filename):
+    return send_from_directory('dist', filename)
+
+@app.route('/api/objects/<object_identifier>/coverage', methods=['GET'])
+def get_coverage_window(object_identifier):
+    coverage_window = {}
+    NAIF_id = None
+    try:
+        NAIF_id = int(object_identifier)
+        if NAIF_id != main_subject:
+            try:
+                spyce.id_to_str(NAIF_id)
+            except spyce.IDNotFoundError:
+                abort(404, "Invalid object id")
+    except ValueError:
+        #object_identifier is name.
+        pass
+
+    if NAIF_id == None:
+        name = object_identifier
+        if name == main_subject_name:
+            NAIF_id = main_subject
+        else:
+            try:
+                NAIF_id = spyce.str_to_id(name)
+            except spyce.IDNotFoundError:
+                abort(404, "Invalid object name")
+
+    windows_piecewise = []
+    for k in kernels:
+        spy.main_file = k
+        try:
+            windows_piecewise += spy.get_coverage_windows(NAIF_id)
+            windows_piecewise.sort(key=lambda x: x[0])
+        except spyce.InternalError:
+            #object does not exist in this kernel.
+            pass
+    coverage_window['start'] = windows_piecewise[0][0]
```

Comment:
you catch the internal error and pass which is fine, but then go on to operate on a potentially empty array. 
Passing 399 to the function throws an IndexError.
check `len(windows_piecewise) > 0`